index.js

```diff
@@ -24,128 +21,134 @@ const { resolveEslintConfig } = require('./lib/resolve-eslint-config')
+    if (!cb) throw new Error('callback is required')
+
+    let result
+    try {
+      result = this.lintTextSync(text, opts)
+    } catch (err) {
+      return process.nextTick(cb, err)
+    }
+    process.nextTick(cb, null, result)
   }
-  process.nextTick(cb, null, result)
-}
-
-/**
- * Lint files to enforce JavaScript Style.
- *
- * @param {Array.<string>} files          file globs to lint
- * @param {BaseLintOptions & { cwd?: string }} [opts] base options + file globs to ignore (has sane defaults) + current working directory (default: process.cwd())
- * @param {LinterCallback} [cb]
- * @returns {void}
- */
-Linter.prototype.lintFiles = function (files, opts, cb) {
-  if (typeof opts === 'function') return this.lintFiles(files, undefined, opts)
-  if (!cb) throw new Error('callback is required')
-
-  const eslintConfig = this.resolveEslintConfig(opts)
-
-  if (typeof files === 'string') files = [files]
-  if (files.length === 0) files = ['.']
 
-  let result
-  try {
-    result = new this.eslint.CLIEngine(eslintConfig).executeOnFiles(files)
-  } catch (err) {
-    return cb(err)
+  /**
+   * Lint files to enforce JavaScript Style.
+   *
+   * @param {Array.<string>} files          file globs to lint
+   * @param {BaseLintOptions & { cwd?: string }} [opts] base options + file globs to ignore (has sane defaults) + current working directory (default: process.cwd())
+   * @param {LinterCallback} [cb]
+   * @returns {void}
+   */
+  lintFiles (files, opts, cb) {
+    if (typeof opts === 'function') { return this.lintFiles(files, undefined, opts) }
+    if (!cb) throw new Error('callback is required')
+
+    const eslintConfig = this.resolveEslintConfig(opts)
+
+    if (typeof files === 'string') files = [files]
+    if (files.length === 0) files = ['.']
+
+    let result
+    try {
+      result = new this.eslint.CLIEngine(eslintConfig).executeOnFiles(files)
+    } catch (err) {
+      return cb(err)
+    }
+
+    if (eslintConfig.fix) {
+      this.eslint.CLIEngine.outputFixes(result)
+    }
+
+    return cb(null, result)
   }
 
-  if (eslintConfig.fix) {
-    this.eslint.CLIEngine.outputFixes(result)
+  /**
+   * @param {BaseLintOptions & { cwd?: string }} [opts]
+   * @returns {CLIEngineOptions}
+   */
+  resolveEslintConfig (opts) {
+    const eslintConfig = resolveEslintConfig(
+      {
+        cwd: this.cwd,
+        ...opts,
+        cmd: this.cmd
+      },
+      this.eslintConfig,
+      this.customEslintConfigResolver
+    )
+
+    return eslintConfig
   }
-
-  return cb(null, result)
 }
 
-/**
- * @param {BaseLintOptions & { cwd?: string }} [opts]
- * @returns {CLIEngineOptions}
- */
-Linter.prototype.resolveEslintConfig = function (opts) {
-  const eslintConfig = resolveEslintConfig(
-    {
-      cwd: this.cwd,
-      ...opts,
-      cmd: this.cmd
-    },
-    this.eslintConfig,
-    this.customEslintConfigResolver
-  )
-
-  return eslintConfig
-}
+module.exports.cli = require('./bin/cmd')
+module.exports.Linter = Linter
```

Comment:
This casing change isn't strictly caused by ES6, but I do like it and it should be done eventually.

I would prefer this export:

```suggestion
module.exports = {
  cli: require('./bin/cmd'),
  Linter
};
```

Reply:
Changed the export but without the semicolon, that's why I didn't commit your suggestion. :smile: